### ISSUE TITLE
Orphan checks fail using forwardslash as a path separator on Windows

### ISSUE THREAD

**Issue by nathanjmcdougall** [agent-ecosystem/skill-validator#63]
## What happened?

The orphan checking logic (`CheckOrphanFiles`) gives false positives on Windows due to differences between the native OS file separator (backslash) versus the conventional separator for markdown (forwardslash).

## What did you expect?

I would have expected no warning about unreferenced files in this case, since `references/other.md` is present in the skill; even though it is using forward slashes this should be considered acceptable

## How to reproduce

Set up a dir with a skill like so:

```
.agents/skills/
└── example-skill/
    ├── references/other.md
    └── SKILL.md
```

Where SKILL.md contains:

```
---
name: example-skill
description: Example skill
---
See references/other.md.
```

Then this will fail with a warning:

```bash
skill-validator check --strict .agents/skills
```

Giving the output:

<details>

```
skill-validator check --strict .agents/skills

Validating skill: C:\Users\namc\Temp\faw\.agents\skills\example-skill

Structure
  ✓ SKILL.md found
  ⚠ potentially unreferenced file: references\other.md — agents may not discover this file without an explicit reference in SKILL.md or a referenced file

Frontmatter
  ✓ name: "example-skill" (valid)
  ✓ description: (13 chars)

Markdown
  ✓ no unclosed code fences found

Tokens
  SKILL.md body:        7 tokens
  references\other.md:  0 tokens
  ───────────────────────────────────────
  Total:                7 tokens

Content Analysis
  Word count:               9
  Code block ratio:         0.00
  Imperative ratio:         0.00
  Information density:      0.00
  Instruction specificity:  0.00
  Sections: 0  |  List items: 0  |  Code blocks: 0

References Content Analysis
  Word count:               0
  Code block ratio:         0.00
  Imperative ratio:         0.00
  Information density:      0.00
  Instruction specificity:  0.00
  Sections: 0  |  List items: 0  |  Code blocks: 0

Contamination Analysis
  Contamination level: low (score: 0.00)
  Scope breadth: 0

References Contamination Analysis
  Contamination level: low (score: 0.00)
  Scope breadth: 0

Result: 1 warning

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

1 skill validated: all passed
Total: 1 warning
```

</details>

## Environment

- skill-validator version: v1.5.2 (commit `1fe10b0`)
- Go version: o version go1.25.9 windows/amd64
- OS: Windows 11 Enterprise 24H2

**Comment by dacharyc** [comment]
Thanks for filing this issue, @nathanjmcdougall  - you're right, I wasn't handling file paths on Windows. I've just pushed an update, v1.5.4, which should fix this issue, along with a handful of other Windows-specific issues I have discovered after setting up Windows in CI.

As a longtime macOS user, this was a complete oversight on my part. Now I'm going to do a little audit of my projects and see what other Windows bugs may be lurking. Thanks for bringing this to my attention!

**Comment by dacharyc** [comment]
Hey @nathanjmcdougall - I'm going to go ahead and close this issue as it's been a few days, and I _think_ it should be resolved now. If you find there are still problems, please feel free to re-open!

**Comment by nathanjmcdougall** [comment]
Yes - it's totally resolved from my perspective. I really appreciated the quick response by the way! This is a super useful tool, I'm really grateful for your work on it.
